### ISSUE TITLE
Questions regarding training strategies, model structure, and loss function

### ISSUE THREAD

**Issue by Alexethz** [billhhh/ShaSpec#4]
Thank you for your great work! I have several questions regarding your research.
1. during training, what is the recommended ratio of complete observation (i.e., all modalities are available) and incomplete observations (i.e., some modalities are missing) if I wanna apply your code to my own dataset?
2. in your code, the first available modality was set as the general shared feature map instead of the average of all available shared feature maps. I wonder if this option does not influence the results so both are fine.
3. in train_SS.py mode_split was returned from the model but not utilized later. Based on your paper, it might be used to omit the loss for the missing modality but this operation was not reflected in the code. Should I follow the code implementation or the paper?

**Comment by billhhh** [comment]
Thank you for your interest in our project. For your questions:

1. At the begining of training, we better have more information of multimodal data to initilize the model to let it grasp the Shared and Specific features among modalities. If in your data, it naturally has small percentage of missing modality (say within 1-5% of data has missing modality and few cases have just 1 modality) instead of severely missing, the model should work fine. You can try to let it train with your missing modal data. But if say over 90% cases just have single modality, then we need the full modality warmup. W.r.t. the ratio, it depends on your task. I would start with 1:10, something like that.
2. Yes, after the cross-modal DAO process, the shared featuref of multimodality should be very close already. So, both average or just use the first available one should work.
3. Again, both would work. As after the cross-modal DAO training, the shared features of each modality will be draged towards each other for a well-alignment. If say we replace the the missing ones with average of the available modality, it just compute the shared features of each avilable modality with this averaged center in the feature space with the loss provided in the code. We tried, both of them received similar results. So I just keep simplicity to implement as the loss in the code.

Hope the answers help!
Best regards

**Comment by billhhh** [comment]
I close the issue for now. If you have more questions, feel free to ask
